### PR TITLE
added files for interacting with books

### PR DESCRIPTION
**Pull request by roman-hrechanyi**

### CODE REVIEW

File: src/main/java/com/grech/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
+    private String title;
+    @NonNull
+    private String author;
+    @NonNull
+    private String isbn;
+    @NonNull
+    @Column(unique = true)
```
Comment (Rommelua) [inline src/main/java/com/grech/model/Book.java:27]: isbn is unique , not a price

File: src/main/java/com/grech/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import javax.sql.DataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@ComponentScan(basePackages = "com.grech")
```
Comment (sokimaaa) [inline src/main/java/com/grech/config/AppConfig.java:11]:
```suggestion
```

File: src/main/java/com/grech/config/AppConfig.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @Autowired
+    public AppConfig(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Bean(name = "entityManagerFactory")
+    public LocalSessionFactoryBean getSessionFactory() {
+        LocalSessionFactoryBean localSessionFactoryBean = new LocalSessionFactoryBean();
+        localSessionFactoryBean.setDataSource(dataSource);
+        localSessionFactoryBean.setPackagesToScan("com.grech.Model");
```
Comment (sokimaaa) [inline src/main/java/com/grech/config/AppConfig.java:24]:
```suggestion
        localSessionFactoryBean.setPackagesToScan("com.grech.model");
```

File: src/main/java/com/grech/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.NonNull;
+
+@Entity
+@Data
+@NoArgsConstructor
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
```
Comment (sokimaaa) [inline src/main/java/com/grech/model/Book.java:20]: @NonNull - annotation for validation
@Column(nullable = false) - annotation for database constraints

you are required to add database constraints, nor validation

File: src/main/java/com/grech/model/Book.java (hunk 1 above)
Comment (sokimaaa) [reply src/main/java/com/grech/model/Book.java:25]: not fixed

File: src/main/java/com/grech/BookShopApplication.java [hunk 5]
```diff
@@ -6,8 +6,8 @@
 @SpringBootApplication
 public class BookShopApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(BookShopApplication.class, args);
-	}
+    public static void main(String[] args) {
+        SpringApplication.run(BookShopApplication.class, args);
+    }
```
Comment (sokimaaa) [inline src/main/java/com/grech/BookShopApplication.java:11]: image

Comment (sokimaaa) [reply src/main/java/com/grech/BookShopApplication.java:11]: not implemented

**Pull request merged**